orc8r/cloud/go/services/configurator/storage/sql_entity_load_helpers.go

```diff
@@ -27,147 +28,195 @@ import (
-	selectBuilder, err := store.getLoadEntitiesSelectBuilder(networkID, filter, criteria)
+
+	rows, err := builder.RunWith(store.tx).Query()
+	if err != nil {
+		return nil, errors.Wrap(err, "error querying for entities")
+	}
+	defer sqorc.CloseRowsLogOnError(rows, "loadEntities")
+
+	for rows.Next() {
+		ent, err := scanEntityRow(rows, criteria)
+		if err != nil {
+			return nil, err
+		}
+		entsByTK[ent.GetTypeAndKey()] = &ent
+	}
+	err = rows.Err()
+	if err != nil {
+		return nil, errors.Wrap(err, "sql rows err")
+	}
+
+	return entsByTK, nil
+}
+
+func (store *sqlConfiguratorStorage) loadAssocs(networkID string, filter EntityLoadFilter, criteria EntityLoadCriteria, loadTyp loadType) (loadedAssocs, error) {
+	if loadTyp != loadChildren && loadTyp != loadParents {
+		return nil, errors.Errorf("wrong load type received: '%v'", loadTyp)
+	}
+
+	assocs := loadedAssocs{}
+
+	builder, err := store.getBuilder(networkID, filter, criteria, loadTyp)
 	if err != nil {
 		return nil, err
 	}
-	rows, err := selectBuilder.RunWith(store.tx).Query()
+
+	rows, err := builder.RunWith(store.tx).Query()
 	if err != nil {
 		return nil, errors.Wrap(err, "error querying for entities")
 	}
-	defer sqorc.CloseRowsLogOnError(rows, "loadFromEntitiesTable")
+	defer sqorc.CloseRowsLogOnError(rows, "loadAssocs")
 
 	for rows.Next() {
-		err = scanNextEntityRow(rows, criteria, entsByPk)
+		a, err := scanAssocRow(rows, loadTyp)
 		if err != nil {
 			return nil, err
 		}
+		assocs = append(assocs, a)
 	}
 	err = rows.Err()
 	if err != nil {
 		return nil, errors.Wrap(err, "sql rows err")
 	}
-	return entsByPk, nil
+
+	return assocs, nil
 }
 
-func (store *sqlConfiguratorStorage) getLoadEntitiesSelectBuilder(networkID string, filter EntityLoadFilter, criteria EntityLoadCriteria) (sq.SelectBuilder, error) {
-	// SELECT ent.pk, ent.key, ent.type, ent.physical_id, ent.version, graph.graph_id, ent.name, ent.description, ent.config,
-	// [[ acl.id, acl.scope, acl.permission, acl.type, acl.id_filter, acl.version ]]
+func (store *sqlConfiguratorStorage) getBuilder(networkID string, filter EntityLoadFilter, criteria EntityLoadCriteria, loadTyp loadType) (sq.SelectBuilder, error) {
+	// Something like:
+	//
+	// SELECT ent.pk, ent.key, ent.type, ent.physical_id, ent.version, ent.graph_id, ent.name, ent.description, ent.config
 	// FROM cfg_entities AS ent
-	// [[ LEFT JOIN cfg_acls AS acl ON acl.entity_pk = ent.pk ]]
-	// [[ WHERE (ent.network_id = $1 AND ent.key = $2 AND ent.type = $3) OR (ent.network_id ...) ... ]]
-	// [[ ORDER BY ent.key ASC LIMIT page_size ]]
-	selectBuilder := store.builder.Select(getLoadEntitiesColumns(criteria)...).
-		From(fmt.Sprintf("%s AS ent", entityTable))
+	// [[ JOIN (on cfg_assocs and cfg_entities to get child/parent assocs) ]]
+	// [[ WHERE ent.network_id = $network_filter AND ent.key = $key_filter AND ent.type = $type_filter AND ent.key > $page_token ]]
+	// ORDER BY ent.key
+	// LIMIT $page_size ;
+
+	// Helpers
+	entCol := func(c string) string {
```

Comment:
Older code seems to be ordering ASC, we aren't doing that now. Is this intentional ?

Reply:
+1 for this

Reply:
ASC is default, both for postgres and maria, so it's unnecessary